scripts/agent-evals/src/mock/mock-tools-main.ts

```diff
@@ -0,0 +1,84 @@
+import { Module } from "module";
+import path from "path";
+import fs from "fs";
+import os from "os";
+import { getFirebaseCliRoot } from "../runner/paths.js";
+import { getToolMocks } from "./tool-mocks.js";
+
+//
+// This file is run as a node --import parameter before the Firebase CLI to
+// patch the implementation for tools with the mocked implementation
+//
+
+// Path to the built MCP Tools implementation in the Firebase CLI, relative to
+// the repo's root
+const MCP_TOOLS_INDEX_PATH = "lib/mcp/tools/index.js";
+const LOG_FILE_PATH = path.join(os.homedir(), "Desktop", "agent_evals_mock_logs.txt");
+// Enable this to turn on file logging. This can be helpful for debugging
+// because console logs get swallowed
+const ENABLE_FILE_LOGGING = true;
+
+const mocks = getToolMocks();
+
+const originalRequire = Module.prototype.require;
+Module.prototype.require = function (id: string) {
+  const requiredModule = originalRequire.apply(this, [id]);
+  const absolutePath = Module.createRequire(this.filename).resolve(id);
+  const pathRelativeToCliRoot = path.relative(getFirebaseCliRoot(), absolutePath);
+  if (!pathRelativeToCliRoot.endsWith(MCP_TOOLS_INDEX_PATH)) {
+    return requiredModule;
+  }
+
+  logToFile(
+    `Creating proxy implementation for file: ${pathRelativeToCliRoot} with tool mocks: ${JSON.stringify(Object.keys(mocks))}`,
+  );
+
+  return new Proxy(requiredModule, {
+    get(target, prop, receiver) {
```

Comment:
Can this be async get? That would let us use await inside instead of old school .then()

Reply:
Ah good call - I can't async this `get` method but I can `return async...` in here. Updated